### PR TITLE
hardening/1049_detect_invalid_configuration_in_grouping_rules

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1049 
* 100% Unit tests passed: 
```
Results : Tests run: 93, Failures: 0, Errors: 0, Skipped: 0
```
* e2e (unofficial) tests passed:
```
[Using agent without grouping interceptor file]
time=2016-05-26T08:40:35.412WEST | lvl=DEBUG | corr= | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.NGSIGroupingInterceptor$Builder[201] : [gi] Grouping rules file cannot be empty

[Using agent with empty grouping interceptor file]
time=2016-05-26T08:48:41.798WEST | lvl=DEBUG | corr= | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.NGSIGroupingInterceptor$Builder[201] : [gi] Grouping rules file cannot be empty
```
* Tests implemented:
```
Running com.telefonica.iot.cygnus.interceptors.NGSIGroupingInterceptorTest
[GroupingInterceptor.Builder.configure] --------- Configured 'grouping_rules_conf_file' cannot be empty
[GroupingInterceptor.Builder.configure] --  OK  - Empty 'grouping_rules_conf_file' has been detected
[GroupingInterceptor.Builder.configure] --------- Configured 'grouping_rules_conf_file' cannot be null
[GroupingInterceptor.Builder.configure] --  OK  - Null 'grouping_rules_conf_file' has been detected
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.082 sec
```
* Assignee: @frbattid

### CODE REVIEW

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 1]
```diff
@@ -44,13 +44,16 @@
     private final String groupingRulesFileName;
     private CygnusGroupingRules groupingRules;
     private ConfigurationReader configurationReader;
+    private Boolean invalidConfiguration = false;
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: boolean

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 2]
```diff
@@ -44,13 +44,16 @@
...
     
     /**
      * Constructor.
      * @param groupingRulesFileName
+     * @param invalidConfiguration
      */
-    public NGSIGroupingInterceptor(String groupingRulesFileName) {
+    public NGSIGroupingInterceptor(String groupingRulesFileName, Boolean invalidConfiguration) {
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: boolean

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 3]
```diff
@@ -63,113 +66,132 @@ protected CygnusGroupingRules getGroupingRules() {
...
-                    + e.getMessage());
-        } // try catch
     } // close
  
     /**
      * Builder class for this new interceptor.
      */
     public static class Builder implements Interceptor.Builder {
         private String groupingRulesFileName;
+        private Boolean invalidConfiguration;
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: boolean

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 4]
```diff
@@ -179,15 +201,20 @@ public void configure(Context context) {
...
             } // if else
         } // configure
  
         @Override
         public Interceptor build() {
-            return new NGSIGroupingInterceptor(groupingRulesFileName);
+            return new NGSIGroupingInterceptor(groupingRulesFileName, invalidConfiguration);
         } // build
+        
+        protected Boolean getInvalidConfiguration() {
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: boolean

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 1 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in 460a0d5

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 2 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in 460a0d5

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in 460a0d5

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 4 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in 460a0d5

**Comment by frbattid** [comment]
LGTM

**Pull request merged**